core-bundle/assets/controllers/autofocus-controller.js

```diff
@@ -0,0 +1,29 @@
+import { Controller } from '@hotwired/stimulus';
+
+export default class extends Controller {
+    static targets = ['input']
+
+    connect () {
+        this.inputTargets.every(input => {
+            if (input.disabled || input.readonly) {
+                return true;
+            }
+
+            if (!input.offsetWidth || !input.offsetHeight) {
+                return true;
+            }
+
+            if (input.closest('.chzn-search')) {
+                return true;
+            }
+
+            if (input.autocomplete && input.autocomplete !== 'off') {
+                return true;
+            }
```

Comment:
I think these checks should als be done on the server side and not add the controller target then. It would then simply be a `this.inputTarget.focus();` (taking the first target).

Reply:
How would you determine `offsetWidth` or `chzn-search` on the server side?

Reply:
Don‘t get me wrong, I would love being able to check this on the server side, because then we could use the `autofocus` attribute and would not need a script at all.

Reply:
`chzn-search` could be checked whether `chosen` is enabled for the given widget I guess - but that still leaves the `offsetWidth` (= visibility) checks.